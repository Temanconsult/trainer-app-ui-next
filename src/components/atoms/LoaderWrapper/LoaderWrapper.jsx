import React from 'react'
import PropTypes from 'prop-types'

const LoaderWrapper = ({ loading, children }) => {
  return loading ? (
    <div className="lds-ellipsis">
      <div />
      <div />
      <div />
      <div />
    </div>
  ) : (
    <>{children}</>
  )
}

LoaderWrapper.propTypes = {
  loading: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.elementType,
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
}

LoaderWrapper.defaultProps = {
  loading: false,
}

export default LoaderWrapper
