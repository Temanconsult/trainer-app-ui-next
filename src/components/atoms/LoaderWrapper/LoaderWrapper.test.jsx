import React from 'react'
import { render } from '@testing-library/react'
import LoaderWrapper from '.'

describe('LoaderWrapper', () => {
  it('creates loader wrapper loading state', () => {
    const { asFragment } = render(
      <LoaderWrapper loading>
        <div>Loading </div>
      </LoaderWrapper>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('creates loader wrapper none loading state', () => {
    const { asFragment } = render(
      <LoaderWrapper>
        <div>Loaded </div>
      </LoaderWrapper>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
