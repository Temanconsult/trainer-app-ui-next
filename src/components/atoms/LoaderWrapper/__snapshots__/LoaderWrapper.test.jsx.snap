// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderWrapper creates loader wrapper loading state 1`] = `
<DocumentFragment>
  <div
    class="lds-ellipsis"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;

exports[`LoaderWrapper creates loader wrapper none loading state 1`] = `
<DocumentFragment>
  <div>
    Loaded 
  </div>
</DocumentFragment>
`;
